Assignment6.Rmd

```diff
@@ -77,10 +96,64 @@ table(M2$certified, M2$predict2)
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
+library(readr)
+
+df<-read.delim("student.course.txt",stringsAsFactors = FALSE,sep = ",")
+
+#create training data and testing data
+set.seed(726)
+df<-df[sample(nrow(df),1000),]
+
+#first model built with raw variable
+model1<-rpart(GPAO~.,df,method="anova")
+
+printcp(model1)
+
+plotcp(model1)
+
+rpart.plot::rpart.plot(model1)
+
+df$predicted1<-predict(model1,df,type = "vector")
+
+ # the mean difference between predicted and actual outcome
+mean(abs(df$predicted1-df$GPAO)) 
+```
+ a mean difference of 0.299  in GPAO from model1's prediction
+```{r}
+#second model built with feature selection or feature extraction
+#using PCA on all the numeric varibles
+df_num<-df[,c(1,3:5,7,8)]
+df_num<-sapply(df_num,as.numeric)
+
+df_num<-data.frame(scale(df_num))
+  
+pca <- prcomp(df_num, scale = TRUE)
+#plot scree plot to decide how many pcs that could be used in the model 
+plot(pca,type="lines")
 
 ```
 
+```{R}
+# according to the result, i decided to keep PC1,PC2
+pca1<-data.frame(pca[["x"]])
+
+dff<-cbind(df,pca1)
+
 
-### To Submit Your Assignment
+model2<-rpart(GPAO~PC1+PC2,dff, method = "anova")
+
+printcp(model2)
+
+plotcp(model2)
+
+rpart.plot::rpart.plot(model2)
+
+dff$predicted2<-predict(model2,dff,"vector")
+mean(abs(dff$predicted2-dff$GPAO))
+
+
+```
 
-Please submit your assignment by first "knitting" your RMarkdown document into an html file and then commit, push and pull request both the RMarkdown file and the html file.
\ No newline at end of file
+ a mean difference of 0.216 in GPAO from model2's prediction
+ 
+ Therefore, model2 did a better job in prediction
```

Comment:
What does this suggest about the second model's generalizability? Great job overall! Keep up the good work.